lib/jekyll/filters.rb

```diff
@@ -389,7 +389,11 @@ def groupable?(element)
     private
     def item_property(item, property)
       if item.respond_to?(:to_liquid)
-        item.to_liquid[property.to_s]
+        res = item.to_liquid
+        property.to_s.split(".").each do |attr|
+          res = res[attr]
+        end
+        res
       elsif item.respond_to?(:data)
         item.data[property.to_s]
```

Comment:
@ashmaroli Also, I'm trying to find in which use case the `else` condition is being called (where the input does not have `.data` or `.to_liquid`) but I can't find.

May it be possible to identify tests going though thanks to code climate ? I didn't find how I can achieve that.

Thanks !